ONSdigital/eq-messaging
# Add account lockout improve password policy

**What**

This PR creates a new role within this ansible repo, allowing a
provisioned machine to forward a set of logs onto a centralised rsyslogd
server. This can be illustrated and tested by the additional changes to
the vagrantfile to test the correct routing of messages to the server.

We also fix a number of related security issues:
- Increase password strength (we don't ever use passwords, only ssh key pairs)
- Increase the failed login block to restrict brute force attacks.
- Add PAM setup to log and forward all user access and escalations to rsyslog.

**How to test**

_Pre-test_
1. Make sure you have tcpdump installed: `brew install tcpdump` 

**Test**
1. Check out this branch
2. cd into the ansible folder
3. Run `vagrant up`
4. This will provision and create a new ubuntu machine with a rsyslogd server assumed to live on your machine.
5. In a new shell run: `sudo tcpdump -i vboxnet2 -X -v 'udp port 514'`
6. In another shell run `vagrant ssh` and then proceed to become root via `sudo su`
7. Check back to your tcpdump shell - you should see raw UDP packets describing your permission escalation to root.

_WARNING_ - This is a breaking change and requires a change to eq-teraform before merging. Please review and then speak to @dhilton to make sure we're good to merge.

**Who can review**

Anyone but @dhilton
